File: src/chipschallenge/clonebehaviors/NullCloneBehavior.java

package chipschallenge.clonebehaviors;

import chipschallenge.Block;


public class NullCloneBehavior extends CloneBehavior {

    private NullCloneBehavior() {
    }
    private static NullCloneBehavior mInstance = null;

    public static synchronized NullCloneBehavior getInstance() {
        if (mInstance == null) {
            mInstance = new NullCloneBehavior();
        }
        return mInstance;
    }

    public Block cloneIt(Block original) throws CloneNotSupportedException {
        throw new CloneNotSupportedException();
    }
}

File: ChipsChallenge.Shared/Clonebehaviors/NullCloneBehavior.cs
namespace ChipsChallenge.Shared.Clonebehaviors
{
    public class NullCloneBehavior : CloneBehavior
    {
        private static NullCloneBehavior instance;

        private NullCloneBehavior()
        {
        }

        public static NullCloneBehavior Instance
        {
            get
            {
                lock (typeof(NullCloneBehavior))
                {
                    return instance ?? (instance = new NullCloneBehavior());
                }
            }
        }

        public override Block CloneIt(Block original)
        {
            throw new CloneNotSupportedException();
        }
    }
}